packages/expo-cli/src/commands/eas-build/utils/expoUpdates.ts

```diff
@@ -0,0 +1,387 @@
+import figures from 'figures';
+import * as fs from 'fs-extra';
+import glob from 'glob';
+import ora from 'ora';
+import path from 'path';
+import xcode from 'xcode';
+
+import { gitAddAsync } from '../../../git';
+import log from '../../../log';
+import * as gitUtils from './git';
+
+const iOSBuildScript = '../node_modules/expo-updates/scripts/create-manifest-ios.sh';
+const androidBuildScript =
+  'apply from: "../../node_modules/expo-updates/scripts/create-manifest-android.gradle"';
+
+export async function isUpdatesConfigured(projectDir: string): Promise<boolean> {
+  if (!isExpoUpdatesInstalled(projectDir)) {
+    return true;
+  }
+
+  const options = await getConfigurationOptions(projectDir);
+
+  return (
+    (await isUpdatesConfiguredAndroid(projectDir, ...options)) &&
+    (await isUpdatesConfiguredIOS(projectDir, ...options))
+  );
+}
+
+export async function configureUpdatesAsync({
+  projectDir,
+  nonInteractive,
+}: {
+  projectDir: string;
+  nonInteractive: boolean;
+}) {
+  if (!isExpoUpdatesInstalled(projectDir)) {
+    return;
+  }
+
+  const spinner = ora('Configuring expo-updates');
+
+  const options = await getConfigurationOptions(projectDir);
+
+  await configureUpdatesAndroid(projectDir, ...options);
+  await configureUpdatesIOS(projectDir, ...options);
+
+  try {
+    await gitUtils.ensureGitStatusIsCleanAsync();
+    spinner.succeed();
+  } catch (err) {
+    if (err instanceof gitUtils.DirtyGitTreeError) {
+      spinner.succeed(`We configured expo-updates in your project `);
+      log.newLine();
+
+      try {
+        await gitUtils.reviewAndCommitChangesAsync(`Configure expo-updates`, { nonInteractive });
+
+        log(`${chalk.green(figures.tick)} Successfully committed the configuration changes.`);
+      } catch (e) {
+        throw new Error(
+          "Aborting, run the command again once you're ready. Make sure to commit any changes you've made."
+        );
+      }
+    } else {
+      spinner.fail();
+      throw err;
+    }
+  }
+}
+
+async function getConfigurationOptions(
+  projectDir: string
+): Promise<[ExpoConfig, string | null, string | null]> {
+  const user = await UserManager.ensureLoggedInAsync();
+
+  const { exp } = getConfig(projectDir);
+
+  if (exp.runtimeVersion) {
+    return [exp, user.username, null];
+  }
+
+  try {
+    const sdkVersion = getExpoSDKVersion(projectDir, exp);
+
+    return [exp, user.username, sdkVersion];
+  } catch (err) {
+    throw new Error(
+      "Couldn't find either 'runtimeVersion' or 'sdkVersion' to configure 'expo-updates'. Please specify at least one of these properties under the 'expo' key in 'app.json'"
+    );
+  }
+}
+
+function isExpoUpdatesInstalled(projectDir: string) {
+  const packageJson = getPackageJson(projectDir);
+
+  return packageJson.dependencies && 'expo-updates' in packageJson.dependencies;
+}
+
+async function configureUpdatesIOS(
+  projectDir: string,
+  exp: ExpoConfig,
+  username: string | null,
+  sdkVersion: string | null
```

Comment:
- sdkVersion is in exp object
- you can read username anywhere without passing it through arguments (you can leave this I don't have a strong opinion on that)